### PR TITLE
FirebaseDeeplink: Handle deeplinks in firebase push notifications

### PR DESCRIPTION
**Pull request by reyhanlioglu**
## Summary by CodeRabbit

- **New Features**
	- Enhanced push notifications with deeplink support for improved navigation.
	- Added a new parameter to customize deeplink navigation behavior.

### CODE REVIEW

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart [hunk 1]
```diff
@@ -25,12 +26,14 @@ class NeoCoreFirebaseMessaging extends StatefulWidget {
     required this.child,
     required this.networkManager,
     this.androidDefaultIcon,
+    this.onDeeplinkNavigation,
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:29]: it's okay to set it nullable

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart [hunk 2]
```diff
@@ -76,25 +80,23 @@ class _NeoCoreFirebaseMessagingState extends State<NeoCoreFirebaseMessaging> {
   }
 
   void _onTokenChange(String token) {
+    print('FCM Token is $token');
```
Comment (mcanyigincommencis) [inline lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:83]: if debug ekleyelim mi

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart [hunk 3]
```diff
@@ -13,10 +13,12 @@ abstract class _Constant {
   static const androidNotificationChannelID = "high_importance_channel";
   static const androidNotificationChannelName = "High Importance Notifications";
   static const androidNotificationChannelDescription = "This channel is used for important notifications";
+  static const pushNotificationDeeplinkKey = "deeplink";
 }
 
 @pragma('vm:entry-point')
-Future<void> onBackgroundMessage(RemoteMessage message) async {
+Future<void> onBackgroundMessage(RemoteMessage message, {Function(RemoteMessage? message)? handleMessage}) async {
```
Comment (mcanyigincommencis) [inline lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:20]: bunun imzasini degistiriyorsun sorun olmayacak mi

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart [hunk 4]
```diff
@@ -76,25 +80,23 @@ class _NeoCoreFirebaseMessagingState extends State<NeoCoreFirebaseMessaging> {
...
+    print('FCM Token is $token');
     NeoCoreRegisterDeviceUseCase().call(networkManager: widget.networkManager, deviceToken: token);
   }
 
   Future<void> _initPushNotifications() async {
     await _firebaseMessaging.setForegroundNotificationPresentationOptions(alert: true, badge: true, sound: true);
 
-    // Get any messages which caused the application to open from
-    // a terminated state.
-    await _firebaseMessaging.getInitialMessage().then(_handleMessage);
```
Comment (mcanyigincommencis) [inline lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:87]: bunu niye sildik

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart [hunk 5]
```diff
@@ -76,25 +80,23 @@ class _NeoCoreFirebaseMessagingState extends State<NeoCoreFirebaseMessaging> {
...
-    // Also handle any interaction when the app is in the background via a
-    // Stream listener
+    // Also handle any interaction when the app is in the background via Stream listener
     FirebaseMessaging.onMessageOpenedApp.listen(_handleMessage);
 
-    ///Sadece Android'de çalışan notification alındığı zaman payload'u alan fonksiyon
-    FirebaseMessaging.onBackgroundMessage(onBackgroundMessage);
+    // Background message handler for Android platform
+    FirebaseMessaging.onBackgroundMessage((message) {
+      print('Background message is $message');
```
Comment (mcanyigincommencis) [inline lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:95]: if debug

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart [hunk 6]
```diff
@@ -82,19 +85,13 @@ class _NeoCoreFirebaseMessagingState extends State<NeoCoreFirebaseMessaging> {
...
-    // a terminated state.
-    await _firebaseMessaging.getInitialMessage().then(_handleMessage);
-
-    // Also handle any interaction when the app is in the background via a
-    // Stream listener
+    // Also handle any interaction when the app is in the background via Stream listener
     FirebaseMessaging.onMessageOpenedApp.listen(_handleMessage);
 
-    ///Sadece Android'de çalışan notification alındığı zaman payload'u alan fonksiyon
+    // Background message handler for Android platform
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:94]: This line handles it 
await _firebaseMessaging.getInitialMessage().then(_handleMessage);

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart (hunk 4 above)
Comment (reyhanlioglu) [reply lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart:87]: tekrar ekledim test ediyordum

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart (hunk 5 above)
Comment (reyhanlioglu) [reply lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart]: deleted

File: lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart (hunk 2 above)
Comment (reyhanlioglu) [reply lib/core/widgets/neo_core_firebase_messaging/neo_core_firebase_messaging.dart]: test amaçlı eklemiştim sildim

**Review by yildirayyorulmaz — APPROVED** [review]

**Pull request merged**